platform/src/server/http/http_server.ts

```diff
@@ -55,41 +86,89 @@ export class HttpServer {
       return;
     }
 
-    await promisify(this.server.close).call(this.server);
-
+    await this.server.stop();
     this.server = undefined;
   }
 
   private initializeServer(config: HttpConfig) {
-    // TODO This was forced to `any` because of the following problem with the
-    // express declaration file, and can be removed once that is fixed:
-    //
-    // ```
-    // Argument of type 'Application' is not assignable to parameter of type
-    // '((request: IncomingMessage, response: ServerResponse) => void) | undefined'
-    // ```
-    const app = this.app as any;
-
-    if (!config.ssl.enabled) {
-      return http.createServer(app);
-    }
+    const options: ServerOptions = {
+      host: config.host,
+      port: config.port,
+      routes: {
+        cors: config.cors,
+        payload: {
+          maxBytes: config.maxPayload.getValueInBytes(),
+        },
+        validate: {
+          options: {
+            abortEarly: false
+          }
+        }
+      },
+      state: {
+        strictHeader: false
+      },
+    };
+
+    const ssl = config.ssl;
+    if (ssl.enabled) {
+      const tlsOptions: TLSOptions = {
+        ca: config.ssl.certificateAuthorities &&
+          config.ssl.certificateAuthorities.map(caFilePath => readFileSync(caFilePath)),
+
+        key: readFileSync(ssl.key!),
+        cert: readFileSync(ssl.certificate!),
+        passphrase: ssl.keyPassphrase,
 
-    // TODO: add support for `secureOptions`.
-    return https.createServer(
-      {
-        key: readFileSync(config.ssl.key!),
-        cert: readFileSync(config.ssl.certificate!),
-        ca:
-          config.ssl.certificateAuthorities &&
-          config.ssl.certificateAuthorities.map(caFilePath =>
-            readFileSync(caFilePath)
-          ),
-        passphrase: config.ssl.keyPassphrase,
         ciphers: config.ssl.cipherSuites.join(':'),
         // We use the server's cipher order rather than the client's to prevent the BEAST attack.
         honorCipherOrder: true,
-      },
-      app
-    );
+        secureOptions: ssl.getSecureOptions(),
+      };
+
+      // TODO: Hapi types have a typo in `tls` property type definition: `https.RequestOptions` is used instead of
+      // `https.ServerOptions`, and `honorCipherOrder` isn't presented in `https.RequestOptions`.
+      options.tls = tlsOptions as any;
```

Comment:
🤦‍♀️ That's too bad.

Reply:
Yeah, I'll submit PR to DefinitelyTyped later, we've submitted a bunch of patches to DT already as the result of type-augmentation-cleanup in new platform and `kbn/pm`. It usually takes a couple of days for the PR to be reviewed and merged.